com.unity.multiplayer.mlapi/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -658,7 +643,15 @@ internal void OnDespawnObject(NetworkObject networkObject, bool destroyGameObjec
             {
                 if (NetworkManager.PrefabHandler.ContainsHandler(networkObject))
                 {
-                    NetworkManager.PrefabHandler.HandleNetworkPrefabDestroy(networkObject);
+                    if(NetworkManager.PrefabHandler.HandleNetworkPrefabDestroy(networkObject))
+                    {
+                        if (SpawnedObjects.Remove(networkObject.NetworkObjectId))
```

Comment:
Why are we only removing it from the spawned object list if we are destroying it? Probably I'm missing something here

Reply:
See your previous question about why we allow the INetworkPrefabInstanceHandler implementation to specify whether or not it wants to allow a NetworkObject to "really be destroyed" or not.  This assure that if the INetworkPrefabInstanceHandler implementation says "ok, go ahead and let this NetworkObject be destroyed for real" then it really destroys it (and removes it from the SpawnedObjects list).